### ISSUE TITLE
Focal loss with internal softmax

### ISSUE THREAD

**Issue by ed-fish** [sauradip/STALE#17]
F.cross_entropy in PyTorch applies a LogSoftmax operation on its inputs internally. This means that torch.exp(-ce_loss) may be applying the softmax on the values twice when computing the focal loss. I'm not sure if this will have any significant impact but may be something to note.

**Comment by ed-fish** [comment]
On review this does not seem to be an issue.
